IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/WebElement.cs

```diff
@@ -0,0 +1,274 @@
+        public Task<string> Selected => RetryAction("selected");
+
+        Point IWebElement.Location => new Point();
+        public Task<string> Location => RetryAction("location");
+
+        Size IWebElement.Size => new Size();
+        public Task<string> Size => RetryAction("size");
+
+        public WebElement FindElement(By by)
+        {
+            if (!Initialized)
+            {
+                WrappedElement = _Driver.FindElement(by);
+            }
+
+            // Eventually swap this out for our own wait
+            WebDriverWait wait = new WebDriverWait(_Driver, TimeSpan.FromSeconds(5));
+            wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(by));
+
+            return new WebElement(WrappedElement.FindElement(by), by, _Driver);
+        }
+
+        public ReadOnlyCollection<WebElement> FindElements(By by)
+        {
+            if (!Initialized)
+            {
+                WrappedElement = _Driver.FindElement(by);
+            }
+
+            // Eventually swap this out for our own wait
+            WebDriverWait wait = new WebDriverWait(_Driver, TimeSpan.FromSeconds(5));
+            try
+            {
+                wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(by));
+            }
+            catch (WebDriverTimeoutException)
+            {
+                return new ReadOnlyCollection<WebElement>(new List<WebElement>());
+            }
+
+            return
+                new ReadOnlyCollection<WebElement>(
+                    WrappedElement.FindElements(by)
+                        .Select(webElement =>
+                        new WebElement(WrappedElement.FindElement(by), by, _Driver))
+                        .ToList());
+        }
+
+        void IWebElement.Clear()
+        {
+            Clear().GetAwaiter().GetResult();
+        }
+        public Task Clear()
+        {
+            return RetryAction("clear");
+        }
+
+        void IWebElement.SendKeys(string text)
+        {
+            SendKeys(text).ConfigureAwait(false).GetAwaiter().GetResult();
+        }
+
+        public Task SendKeys(string text)
+        {
+            return RetryAction("sendkeys", text);
+        }
+
+        void IWebElement.Submit()
+        {
+            Submit().GetAwaiter().GetResult();
+        }
+
+        public Task Submit()
+        {
+            return RetryAction("submit");
+        }
+
+        void IWebElement.Click()
+        {
+            Click().GetAwaiter().GetResult();
+        }
+
+        public Task Click()
+        {
+            return RetryAction("click");
+        }
+
+        // Can this easily be abstracted out to an extension method?
+        // Seems to make more sense as an extension
+        public void Click(int timeToRetry)
+        {
+            TimeToRetry = timeToRetry;
+            Click();
+            TimeToRetry = DefaultRetrySeconds;
```

Comment:
This setting and resetting `TimeToRetry` is a bit problematic. For one, if this value was set to anything other DefautltRetrySeconds, you will have lost that value after calling this method. In general setting and unsetting values like this is problematic. A better design is to pass the `timeToRetry` into the Click method, and then into the `RetryAction` as an optional parameter (using the TimeToRetry if the optional parameter is not set). This avoids needing to keep track of previous values and reset appropriately.